paimon-core/src/main/java/org/apache/paimon/catalog/AbstractCatalog.java

```diff
@@ -83,6 +86,7 @@ protected AbstractCatalog(FileIO fileIO, Options options) {
         this.tableDefaultOptions =
                 convertToPropertiesPrefixKey(options.toMap(), TABLE_DEFAULT_OPTION_PREFIX);
         this.catalogOptions = options;
+        this.branchName = options.get(CoreOptions.BRANCH);
```

Comment:
What is the scope of branch option?
Catalog? Table? Why catalog need to know?

Reply:
> What is the scope of branch option? Catalog? Table? Why catalog need to know?

The scope is Table, but when using getTable() or manipulating the schema in the catalog, it is necessary to specify the branch.

Reply:
The scope is catalog, please respect your codes.